### ISSUE TITLE
Make class or record sealed - False error when using top-level statements

### ISSUE THREAD

**Issue by Code-Chops** [meziantou/Meziantou.Analyzer#926]
### Version of the Meziantou.Analyzer NuGet package

2.0.252

### Rule Identifier

MA0053

### Target Framework

.net10

### C# Language version

C# 14

### Description

When having a Program.cs with top-level statements, MA0053 gives an error `Error  MA0053 : Make class or record sealed (https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0053.md)`.

This is incorrect, as I cannot seal that class.

### Reproduction Steps

Minimal code:

```c#
code to reproduce the error
```

**Comment by meziantou** [comment]
This is not expected. Can you share a repro (csproj + cs files)?

**Comment by Code-Chops** [comment]
[Repro.zip](https://github.com/user-attachments/files/23525795/Repro.zip)

**Comment by meziantou** [comment]
Sorry I do not repro the error you get. I open your solution in VS2026 and I see nothing in the error window. I also try to build it using `dotnet build` and it doesn't report any warning.

- Do you have any error when building using `dotnet build`?
- Which version of the compiler are you using? (https://www.meziantou.net/getting-roslyn-csharp-compiler-and-language-versions.htm)

Note: `dotnet_diagnostic.CA1816.severity = default` is not a valid value. But it should not have any impact for this issue.

**Comment by Code-Chops** [comment]
Somehow I also don't get the error again. I don't know why. I will close this thread :)
